ci_scripts/build_external_tfm.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+"""
+Copyright (c) 2021 ARM Limited. All rights reserved.
+
+SPDX-License-Identifier: Apache-2.0
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import sys
+import signal
+import logging
+
+sys.path.append("../")
+from psa_builder import *
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[Build-external-TF-M] %(asctime)s: %(message)s.",
+    datefmt="%H:%M:%S",
+)
+
+
+def _build_tfm(args, config, suite=None):
+    """
+    Build TF-M regression test
+    :param args: Command-line arguments
+    :param config: Config type
+    :param suite: Test suite for PSA compliance
+    """
+    cmd = [
+        "python3",
+        "build_tfm.py",
+        "-m",
+        args.mcu,
+        "-t",
+        args.toolchain,
+        "-c",
+        config,
+        "--skip-clone",
+        "--skip-copy",
+    ]
+
+    if config in SUPPORTED_TFM_PSA_CONFIGS:
+        cmd.append("-s")
+        cmd.append(suite)
+
+    retcode = run_cmd_output_realtime(cmd, ROOT)
+    if retcode:
+        logging.critical("Unable to build TF-M for target - %s", args.mcu)
+        sys.exit(1)
+
+
+def _build_regression_test(args):
+    """
+    Build TF-M regression test for the target
+    :param args: Command-line arguments
+    """
+    logging.info("Build TF-M regression tests for %s", args.mcu)
+
+    _build_tfm(args, "RegressionIPC")
```

Comment:
Shall we copy out the binaries after each build, or use `srec_cat` to generate a hex file named after the suite? This would enable the CI to collect artefacts, or us to build and try all tests locally.

Reply:
Are we thinking of testing things locally, then I can add `mbedhtrun` as we did it in https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/blob/c568e4ab5c185190ccd7b8a98efdf5f5d594cc8f/test_psa_target.py

In this way, we can run it locally, but Jenkins CI can just run the build?

Reply:
It would still be good to keep a copy of everything we build. Otherwise the binaries are overridden we move from one suite to another.
A CI run saves the artefacts which we can download.

Reply:
done